### PR TITLE
gef: init at 2022.01

### PR DESCRIPTION
**Pull request by Freax13**
###### Description of changes
Add gef.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/gef/default.nix [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    keystone-engine
+    unicorn
+    capstone
+    ropper
+  ];
+
+in stdenv.mkDerivation rec {
+  pname = "gef";
+  version = "2022.01";
+  format = "other";
```
Comment (lourkeur) [inline pkgs/development/tools/misc/gef/default.nix:24]: Is this needed given that this isn't a `buildPython*` derivation?

Comment (Freax13) [reply pkgs/development/tools/misc/gef/default.nix]: I suppose not.

I started this by copying the `pwndbg` package which also has this line:
https://github.com/NixOS/nixpkgs/blob/a6a8886d70a2b0ccd87e1623b6a4b96c9459a29a/pkgs/development/tools/misc/pwndbg/default.nix#L26

**Comment by fabaff** [comment]
Related to #81418

**Comment by sikmir** [comment]
Just in case, [gef in NUR](https://github.com/nix-community/nur-combined/blob/d629194507acb3a6ffbcce069bfb704e374fbc12/repos/sikmir/pkgs/misc/gef/default.nix),

**Pull request merged**